main.go

```diff
@@ -25,21 +29,20 @@ func validPort(port string) (string, error) {
 	return fmt.Sprintf(":%s", port), nil
 }
 
-func registerHandlers(api *API, mux *http.ServeMux) http.Handler {
-	mux.HandleFunc("/api/scan", apiWrapper(api.Scan))
-	mux.HandleFunc("/api/queue", apiWrapper(api.Queue))
-	mux.HandleFunc("/api/validate", apiWrapper(api.Validate))
-
-	originsOk := handlers.AllowedOrigins([]string{os.Getenv("ALLOWED_ORIGINS")})
-
-	handler := http.HandlerFunc(recoveryHandler(
-		handlers.CORS(originsOk)(mux).ServeHTTP,
-	))
-	return handlers.LoggingHandler(os.Stdout, handler)
+func throttle(period time.Duration, limit int64, f http.Handler) http.Handler {
+	rateLimitStore := memory.NewStore()
+	rate := limiter.Rate{
+		Period: period,
+		Limit:  limit,
+	}
+	rateLimiter := stdlib.NewMiddleware(limiter.New(rateLimitStore, rate),
+		stdlib.WithForwardHeader(true))
```

Comment:
@vbrown608 just confirming-- this allows us to detect IPs even though we're behind a load balancer?

Reply:
This will check the `x-forwarded-for` header, which gets set by the load balancer :)